tools/wptrunner/wptrunner/executors/executoratspi.py

```diff
@@ -25,6 +43,24 @@ def find_active_tab(root):
     return None
 
 
+def is_ready(active_tab, product):
+    # Firefox uses the "BUSY" state to indicate the page is not ready.
+    if product == "firefox":
+        state_set = Atspi.Accessible.get_state_set(active_tab)
+        return not Atspi.StateSet.contains(state_set, Atspi.StateType.BUSY)
+
+    # Chromium family browsers do not use "BUSY", but you can
+    # tell if the document can be queried by Title attribute. If the 'Title'
+    # attribute is not here, we need to query for a new accessible object.
+    # TODO: eventually we should test this against the actual title of the
+    # page being tested.
+    document = Atspi.Accessible.get_document_iface(active_tab)
+    document_attributes = Atspi.Document.get_document_attributes(document)
+    if "Title" in document_attributes and document_attributes["Title"]:
+        return True
+    return False
```

Comment:
Ok so this is a little bit awkward, but polling only for the busy state doesn't work on Chromium family browsers. Additionally you can't just keep testing for the Title attribute, a new accessible object is created when the document is available.

Reply:
Oof, I had a play with this and I see how we ended up here.

This seems about as good as we're likely to get unless we make a change to Chromium to expose the busy state.

I did notice we get a "URI" property as well as a title; I meant to comment on the Windows change that if we can check the URL rather than the title, that'll avoid some potential ambiguity.

Reply:
Good point, I'll switch the window's PR to use the url, and open a new PR to check the URL here at that time.